aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -268,30 +332,46 @@ Tensor empty_sparse_csr(
     c10::optional<bool> pin_memory,
     c10::optional<MemoryFormat> optional_memory_format) {
   check_size_nonnegative(size);
-
-  TORCH_CHECK(size.size() >= 2, "torch.empty: Only batched sparse CSR matrices are supported, but got size ", size);
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(layout == Layout::SparseCsr);
-
-  auto rows = size[size.size() - 2];
-  int64_t nnz = 0;
-
-  auto crow_indices_size = DimVector(size.slice(0, size.size() - 2));
-  crow_indices_size.push_back(rows + 1);
-  auto col_indices_values_size = DimVector(size.slice(0, size.size() - 2));
-  col_indices_values_size.push_back(nnz);
+  Layout layout_ = layout.value_or((size.size() == 2 ? kSparseCsr : kSparseBsr));  // TODO: this is wrong..
```

Comment:
I think it's fair to just assume kSparseCsr unless the layout was explicitly passed.